static/js/components/ShiftPage.jsx

```diff
@@ -28,30 +29,87 @@ function datestringToDate(shiftList) {
   return shiftList;
 }
 
-const ShiftPage = () => {
+const ShiftPage = ({ route }) => {
   const classes = useStyles();
-  let { shiftList } = useSelector((state) => state.shifts);
+  const dispatch = useDispatch();
   const currentUser = useSelector((state) => state.profile);
-  shiftList = datestringToDate(shiftList);
+  const shiftList = useSelector((state) => state.shifts.shiftList);
+  const currentShift = useSelector((state) => state.shift.currentShift);
+  const [events, setEvents] = React.useState([]);
 
-  if (!shiftList) {
-    return <CircularProgress />;
+  if (shiftList) {
+    if (!events) {
+      setEvents(datestringToDate(shiftList));
+    } else if (events.length !== shiftList.length) {
+      setEvents(datestringToDate(shiftList));
+    } else if (currentShift) {
+      if (Object.keys(currentShift).length > 0) {
+        if (
+          events.find((shift) => shift.id === currentShift.id).shift_users
+            .length !== currentShift.shift_users.length
+        ) {
+          setEvents(datestringToDate(shiftList));
+        }
+      }
+    }
```

Comment:
a lot of these clauses end with the same `setEvents`. Can we compress the logic here a bit?